commit 477755128b87289c41a8f7ee1a706e893dff203e
Author: Jonathan Ellis <jbellis@apache.org>
Date:   Mon Dec 14 20:01:49 2009 +0000

    add missing volatile, final modifiers.  patch by jbellis; reviewed by Stu Hood for CASSANDRA-614
    
    git-svn-id: https://svn.apache.org/repos/asf/incubator/cassandra/trunk@890461 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/src/java/org/apache/cassandra/db/HintedHandOffManager.java b/src/java/org/apache/cassandra/db/HintedHandOffManager.java
index 55c0d62..6d72047 100644
--- a/src/java/org/apache/cassandra/db/HintedHandOffManager.java
+++ b/src/java/org/apache/cassandra/db/HintedHandOffManager.java
@@ -74,23 +74,23 @@ import org.apache.cassandra.db.filter.QueryPath;
 
 public class HintedHandOffManager
 {
-    private static HintedHandOffManager instance_;
-    private static Lock lock_ = new ReentrantLock();
-    private static Logger logger_ = Logger.getLogger(HintedHandOffManager.class);
+    private static volatile HintedHandOffManager instance_;
+    private static final Lock lock_ = new ReentrantLock();
+    private static final Logger logger_ = Logger.getLogger(HintedHandOffManager.class);
     final static long INTERVAL_IN_MS = 3600 * 1000;
-    private ExecutorService executor_ = new DebuggableThreadPoolExecutor("HINTED-HANDOFF-POOL");
-    Timer timer = new Timer("HINTED-HANDOFF-TIMER");
+    private final ExecutorService executor_ = new DebuggableThreadPoolExecutor("HINTED-HANDOFF-POOL");
+    final Timer timer = new Timer("HINTED-HANDOFF-TIMER");
     public static final String HINTS_CF = "HintsColumnFamily";
 
 
     public static HintedHandOffManager instance()
     {
-        if ( instance_ == null )
+        if (instance_ == null)
         {
             lock_.lock();
             try
             {
-                if ( instance_ == null )
+                if (instance_ == null)
                     instance_ = new HintedHandOffManager();
             }
             finally
diff --git a/src/java/org/apache/cassandra/service/ReadRepairManager.java b/src/java/org/apache/cassandra/service/ReadRepairManager.java
index 704bfa3..a91f9f6 100644
--- a/src/java/org/apache/cassandra/service/ReadRepairManager.java
+++ b/src/java/org/apache/cassandra/service/ReadRepairManager.java
@@ -47,10 +47,10 @@ import org.apache.log4j.Logger;
  */
 class ReadRepairManager
 {
-    private static Logger logger_ = Logger.getLogger(ReadRepairManager.class);
+    private static final Logger logger_ = Logger.getLogger(ReadRepairManager.class);
 	private static final long expirationTimeInMillis = 2000;
-	private static Lock lock_ = new ReentrantLock();
-	private static ReadRepairManager self_ = null;
+	private static final Lock lock_ = new ReentrantLock();
+	private static volatile ReadRepairManager self_ = null;
 
 	/*
 	 * This is the internal class which actually
@@ -94,7 +94,7 @@ class ReadRepairManager
             lock_.lock();
             try
             {
-                if ( self_ == null )
+                if (self_ == null)
                     self_ = new ReadRepairManager();
             }
             finally
